src/app/CommandPathParams.h

```diff
@@ -0,0 +1,54 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/util/basic-types.h>
+#include <support/BitFlags.h>
+
+namespace chip {
+namespace app {
+
+enum class CommandPathFlags : uint8_t
+{
+    kEndpointIdValid = 0x01,
+    kGroupIdValid    = 0x02,
+};
+
+struct CommandPathParams
+{
+    CommandPathParams(EndpointId aEndpointId, GroupId aGroupId, ClusterId aClusterId, CommandId aCommandId,
+                      const BitFlags<CommandPathFlags> & aFlags) :
+        mEndpointId(aEndpointId),
+        mGroupId(aGroupId), mClusterId(aClusterId), mCommandId(aCommandId), mFlags(aFlags)
+    {}
+    bool IsSamePath(const CommandPathParams & aCommandPathParams) const
+    {
+        return aCommandPathParams.mClusterId == mClusterId && aCommandPathParams.mCommandId == mCommandId &&
+                mFlags.Has(aCommandPathParams.mFlags) && mFlags.Has(CommandPathFlags::kEndpointIdValid)
+            ? aCommandPathParams.mEndpointId == mEndpointId
+            : false || mFlags.Has(CommandPathFlags::kGroupIdValid) ? aCommandPathParams.mGroupId == mGroupId : false;
+    }
+    EndpointId mEndpointId            = 0;
+    GroupId mGroupId                  = 0;
+    ClusterId mClusterId              = 0;
+    CommandId mCommandId              = 0;
+    BitFlags<CommandPathFlags> mFlags = CommandPathFlags::kEndpointIdValid;
```

Comment:
How about initializing this to 0? EndpointId is not yet valid.

Reply:
mFlags can only be initialized with CommandPathFlags::kEndpointIdValid or CommandPathFlags::kGroupIdValid, if you put 0, complication cannot pass

Reply:
just remove intiializer, it is default as 0